global-cli/index.js

```diff
@@ -167,6 +170,29 @@ function checkNodeVersion() {
   }
 }
 
+function checkAppName(appName) {
+  // TODO: there should be a single place that holds the dependencies
+  var dependencies = ['react', 'react-dom'];
+  var devDependencies = ['react-scripts'];
+  var allDependencies = dependencies.concat(devDependencies).sort();
+
+  if (allDependencies.indexOf(appName) >= 0) {
+    console.error(
+      chalk.red(
+        `Can't use "${appName}" as the app name because a dependency with the same name exists.\n\n` +
```

Comment:
Oops, I'm very sorry. I've got confused by [this line](https://github.com/facebookincubator/create-react-app/blob/master/package.json#L8) but now I realized it's for create-react-app project itself.